protocol/error.go

```diff
@@ -1,10 +1,15 @@
 // Defines constants representing the types
-// of errors that the server may return to a client.
+// of errors that the server may return to a client,
+// and that the client may throw after a consistency
+// check or cryptographic verification
 
 package protocol
 
+// An ErrorCode is a special int.
```

Comment:
> they can be mapped to human readable string representations.

I would like to add something like "... it implements the built-in error interfaces".